### PR TITLE
JDBC SNAPSHOT Jars

### PR DESCRIPTION
**Pull request by hannes**
This PR fully automates the JDBC/Maven release process. We also start uploading SNAPSHOT releases to Maven central. I've centralised the JDBC builds into their own separate workflow as well.

### CODE REVIEW

**Comment by hannes** [comment]
Not so sure about the `paths-ignore:` there

**Review by Mytherin — APPROVED** [review]
Thanks! Looks good to me. Good idea to split the Java build into a separate CI workflow.

**Comment by georgewfraser** [comment]
Thank you!

File: scripts/jdbc_maven_deploy.py [hunk 1]
```diff
@@ -108,79 +128,70 @@ def exec(cmd):
...
 exec("jar -cvf %s -C %s/src/main/java org" % (sources_jar, jdbc_root_path))
 
 # make sure all files exist before continuing
 if not os.path.exists(javadoc_jar) or not os.path.exists(sources_jar) or not os.path.exists(pom) or not os.path.exists(binary_jar):
-	raise ValueError('could not create all required files') 
+  raise ValueError('could not create all required files') 
 
 # run basic tests, it should now work on whatever platform this is
-exec("java -cp %s org.duckdb.test.TestDuckDBJDBC" % binary_jar)
+# exec("java -cp %s org.duckdb.test.TestDuckDBJDBC" % binary_jar)
```
Comment (samansmink) [inline scripts/jdbc_maven_deploy.py:151]: should this remain commented out?

Comment (hannes) [reply scripts/jdbc_maven_deploy.py:151]: good catch, let me try

**Pull request merged**